cube/cube.c

```diff
@@ -3178,8 +3179,7 @@ static void demo_init_vk(struct demo *demo) {
         VkPhysicalDevice *physical_devices = malloc(sizeof(VkPhysicalDevice) * gpu_count);
         err = vkEnumeratePhysicalDevices(demo->inst, &gpu_count, physical_devices);
         assert(!err);
-        /* For cube demo we just grab the first physical device */
-        demo->gpu = physical_devices[0];
+        demo->gpu = physical_devices[demo->gpu_number];
```

Comment:
This segfaults if the number selected by the user is out of bounds.

It would be cleaner if it either:
1. Crashed more gracefully with an error message.
2. Output a warning to the user and use the first gpu in the list.